packages/jss/src/index.d.ts

```diff
@@ -7,35 +7,38 @@ import {Properties as CSSProperties} from 'csstype'
 // TODO: refactor to only include Observable types if plugin is installed.
 import {Observable} from 'indefinite-observable'
 
-// TODO: Type data better, currently typed as any for allowing to override it
-type Func<R> = ((data: any) => R)
+type Func<P, T, R> = T extends undefined ? ((data: P) => R) : ((data: P & {theme: T}) => R)
 
-type NormalCssProperties = CSSProperties<string | number>
+type NormalCssProperties = css.Properties<string | number>
 type NormalCssValues<K> = K extends keyof NormalCssProperties
-  ? NormalCssProperties[K]
+  ? NormalCssProperties[K] | JssValue
   : JssValue
 
-export type JssStyle =
+export type JssStyle<Props = unknown, Theme = undefined> =
```

Comment:
Looks like you use this a lot with `any` i think the default should be `any` not `unknown`. What's the behavior if it's `unknown`? Will it break types somewhere?

Reply:
I don't think the types would break. `unknown` is considered the "type-safe" version of `any` because it forces the end user to specify types. And using `any` is typically discouraged. So my brain has defaulted to using `unknown` in place of `any` in consequence.

From the user's perspective, `unknown` will force the user to define types explicitly. So if they want to destructure or use `props`, they'll need to provide the type as a generic (which is a good thing). Additionally, many TS linters will complain if you're using something that implicitly has the `any` type, so in some cases `unknown` may as well be used.

However, looking at `JssStyle`, I think I may agree that it would actually be better to make `Props` default to `any`. As far as hooks/HOCs are concerned, users will only be interfacing with `Styles`. So `Styles` is probably the only type that needs to default to `unknown`.

Reply:
I guess those who didn't specify the theme type will not be able to upgrade because this will require them. How do other TS projects handle such changes, as breaking change and major version increment?

Reply:
@moshest Changed the default to `any`. Mind giving the file a second glance to make sure things still look normal?

Reply:
seems fine now!